fc/code_generation.py

```diff
@@ -59,6 +60,67 @@ def load_template(*name):
     return env.get_template(path)
 
 
+class DataInterpolation(Quantity):
+    """Represents linear interpolation on data loaded from file, for code generation."""
+    _next_id = 0  # Used to generate unique table IDs in code
+
+    # Sympy annoyingly overwrites __new__
+    def __new__(cls, name, data, index_variable, *args, **kwargs):
+        print('interp__new__', name)
+        obj = super().__new__(cls, name, real=True)
+
+        # Record a unique ID for this table
+        obj._id = cls._next_id
+        cls._next_id += 1
+
+        # Ensure we depend on the index variable in the model's graph
+        obj._args = [index_variable]
+
+        return obj
+
+    def __init__(self, name, data, index_variable, units):
+        """Create a new interpolation construct.
+
+        :param name: the data file base name
+        :param data: 2d numpy array containing the column-wise data
```

Comment:
Time/value ? Or can it be vector values too?

Reply:
Index/value; both are vectors. I'll add to the docstring.